### PR TITLE
Added comments to input messages

### PR DESCRIPTION
**Pull request by KipHamiltons**
Let me know if I should add a different reviewer.
Most of this is probably wrong of course.

### CODE REVIEW

File: shared/message/input/GameState.proto [hunk 1]
```diff
@@ -56,32 +73,110 @@ message GameState {
...
+         * Indicator variable describing whose kick off is next.
+         */
+        bool our_kick_off = 6;
+
+        /**
+         * Timestamp when the robot thinks the half is due to end.
+         */
+        google.protobuf.Timestamp primary_time = 7;
+
+        google.protobuf.Timestamp secondary_time = 8;
```
Comment (Bidski) [inline shared/message/input/GameState.proto:158]: What is secondary time?

Comment (TrentHouliston) [reply shared/message/input/GameState.proto:158]: It's a thing in the game controller protocol. E.g. primary time might count the half time, but secondary time will count how long in ready, or how long until penalty etc

File: shared/message/input/GameEvents.proto [hunk 2]
```diff
@@ -23,84 +23,122 @@ package message.input;
...
+        /// The party kicking off
+        Context context = 1;
+    }
+    /// Set of timestamps for different parts of the game, with current phase state
     message GamePhase {
+
+        /// Current phase state, such as PLAYING or TIMEOUT
         GameState.Data.Phase phase = 1;
 
         // INITIAL
```
Comment (ysims) [inline shared/message/input/GameEvents.proto:115]: what are these things?
// INITIAL, // READY, etc

File: shared/message/input/GameState.proto [hunk 3]
```diff
@@ -22,15 +22,22 @@ package message.input;
...
 
+/**
+ * Contains lots of game state information
+ */
 message GameState {
+
+    /// All of the information from GameState except the event string
     message Data {
+
+        /// Type of gameplay - UNKNOWN_MODE, NORMAL, PENALTY_SHOOTOUT or OVERTIME
```
Comment (ysims) [inline shared/message/input/GameState.proto:33]:
```suggestion
        /// Type of gameplay
```
Just seems a little redundant

File: shared/message/input/GameState.proto [hunk 4]
```diff
@@ -55,33 +62,54 @@ message GameState {
...
-            repeated Robot players       = 4;
+            /// The robot which is being tracked
+            uint32 id = 1;
+            /// The infraction the robot being tracked committed
+            PenaltyReason penalty_reason = 2;
+            /// The timestamp when the robot will be playing again
+            google.protobuf.Timestamp unpenalised = 3;
         }
+        /// Set of robots on a team, with a score and coach message
+        message Team {
```
Comment (ysims) [inline shared/message/input/GameState.proto:77]: This newline looks a bit strange to me, it's in a few other places in the proto documentation too.

File: shared/message/input/GameState.proto [hunk 5]
```diff
@@ -55,33 +62,54 @@ message GameState {
...
+        /// The opponent's team - set of players, score, and coach message
+        Team opponent = 10;
+        /// Tracking information about this robot, including penalty reason and unpenalised timestamp
+        Robot self = 11;
     }
+    /// The set of submessages with gamestate information
+    Data data = 1;
 
-    Data   data  = 1;
     string event = 2;
```
Comment (ysims) [inline shared/message/input/GameState.proto:114]: What is `string event`?

File: shared/message/input/Image.proto [hunk 6]
```diff
@@ -25,27 +25,45 @@ import "google/protobuf/timestamp.proto";
...
 
+/**
+ * Camera, lens, and output image information
+ */
 message Image {
+
+    /// The lens type and details about the images it makes
     message Lens {
+
+        /// Types of projection of our lenses
```
Comment (ysims) [inline shared/message/input/Image.proto:36]:
```suggestion
        /// Lens projection types
```

File: shared/message/input/Image.proto [hunk 7]
```diff
@@ -25,27 +25,45 @@ import "google/protobuf/timestamp.proto";
...
             RECTILINEAR = 1;
             EQUIDISTANT = 2;
             EQUISOLID   = 3;
         }
-
-        Projection projection   = 1;
-        float      focal_length = 2;
-        float      fov          = 3;
-        fvec2      centre       = 4;
+        /// The type of projection this lens is using
```
Comment (ysims) [inline shared/message/input/Image.proto:43]:
```suggestion
        /// Lens projection type
```

File: shared/message/input/GameEvents.proto (hunk 2 above)
Comment (KipHamiltons) [reply shared/message/input/GameEvents.proto:115]: No idea

File: shared/message/input/GameState.proto (hunk 5 above)
Comment (KipHamiltons) [reply shared/message/input/GameState.proto:114]: no idea

File: shared/message/input/GameEvents.proto (hunk 2 above)
Comment (KipHamiltons) [reply shared/message/input/GameEvents.proto:115]: Josephus thinks they mean that the variable below each of those comments is set after the gamecontroller has selected those respective states, so we'll run with that unless someone has any other ideas.

Comment (ysims) [reply shared/message/input/GameEvents.proto]: This seems reasonable enough.

**Review by ysims — APPROVED** [review]
Looks fine. I'm okay with leaving `string event` unless @TrentHouliston or @Bidski know what it is? Can't see it being used anywhere.

**Comment by TrentHouliston** [comment]
> Looks fine. I'm okay with leaving `string event` unless @TrentHouliston or @Bidski know what it is? Can't see it being used anywhere.

I can't see it as any part of the specification provided for the game controller https://github.com/RoboCup-Humanoid-TC/GameController/blob/master/include/RoboCupGameControlData.h and if it's not used then I see no reason to keep it

**Comment by KipHamiltons** [comment]
> > Looks fine. I'm okay with leaving `string event` unless @TrentHouliston or @Bidski know what it is? Can't see it being used anywhere.
> 
> I can't see it as any part of the specification provided for the game controller https://github.com/RoboCup-Humanoid-TC/GameController/blob/master/include/RoboCupGameControlData.h and if it's not used then I see no reason to keep it

I have removed it, although I forgot to mention that in the commit message :shrug:

File: shared/message/input/GameState.proto [hunk 8]
```diff
@@ -73,110 +55,34 @@ message GameState {
...
+        Phase                     phase            = 1;
+        Mode                      mode             = 2;
+        bool                      first_half       = 3;
+        bool                      kicked_out_by_us = 4;
+        google.protobuf.Timestamp kicked_out_time  = 5;
+        bool                      our_kick_off     = 6;
+        /// Timestamp when the robot thinks the half is due to end
         google.protobuf.Timestamp primary_time = 7;
-
+        /// A thing in the game controller protocol. E.g. primary time might
```
Comment (Bidski) [inline shared/message/input/GameState.proto:80]: "A thing" feels way too vague for documentation

File: shared/message/input/Image.proto [hunk 9]
```diff
@@ -25,85 +25,34 @@ import "google/protobuf/timestamp.proto";
...
-
-        /**
-         * The type of projection this lens is using.
-         */
         Projection projection = 1;
-
-        /**
-         * Normalised focal length: focal length in pixels / image width.
-         */
+        /// Normalised focal length: focal length in pixels / image width
```
Comment (Bidski) [inline shared/message/input/Image.proto:38]: @TrentHouliston are these still normalised? I thought we stopped using normalised coordinates? Or was that somewhere else?

**Review by Bidski — CHANGES_REQUESTED** [review]

File: shared/message/input/Image.proto (hunk 9 above)
Comment (KipHamiltons) [reply shared/message/input/Image.proto:38]: @TrentHouliston

**Review by Bidski — APPROVED** [review]
Sure

**Pull request merged**